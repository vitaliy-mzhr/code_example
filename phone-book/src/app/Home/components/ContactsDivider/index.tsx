import * as React from 'react';
import ContactsDividerStyles from './styles';
import { Paper, Typography } from '@material-ui/core';

interface AppContactsDividerProps {
  divider: string;
}

class ContactsDivider extends React.PureComponent<AppContactsDividerProps, {}> {
  public render() {
    const {divider} = this.props;
    return (
      <ContactsDividerStyles>
        <Paper className="divider" >
          <Typography variant="headline" component="p" className="title">
            {divider.toUpperCase()}
          </Typography>
        </Paper>
      </ContactsDividerStyles>
    );
  }
}

export default ContactsDivider;
