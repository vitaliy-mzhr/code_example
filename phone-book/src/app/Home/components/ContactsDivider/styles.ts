import styled from 'styled-components';

const ContactsDividerStyles = styled.div`
  .divider {
    padding: .5rem;
    margin-bottom: .25rem;
    background-color: ${props => props.theme.colors.greyColor};
  }
  .title {
    text-align: center;
  }
`;

export default ContactsDividerStyles;
