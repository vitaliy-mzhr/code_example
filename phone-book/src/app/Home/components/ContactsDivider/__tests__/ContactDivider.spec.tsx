import * as React from 'react';
import { shallow } from 'enzyme';
import ContactsDivider from '../../ContactsDivider';

it('ContactsDivider component renders without crashing', () => {
  expect(shallow(<ContactsDivider divider="test"/>)).toMatchSnapshot();
});