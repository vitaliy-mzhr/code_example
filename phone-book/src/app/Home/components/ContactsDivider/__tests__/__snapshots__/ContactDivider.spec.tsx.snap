// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactsDivider component renders without crashing 1`] = `
<styled.div>
  <WithStyles(Paper)
    className="divider"
  >
    <WithStyles(Typography)
      className="title"
      component="p"
      variant="headline"
    >
      TEST
    </WithStyles(Typography)>
  </WithStyles(Paper)>
</styled.div>
`;
